torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1663,6 +1679,19 @@ def _wait_for_previous_optim_step(self) -> None:
 
         self._streams["all_gather"].wait_stream(torch.cuda.current_stream())
 
+    def _need_prefetch_forward_full_params(self) -> bool:
+        if (
+            self.forward_prefetch.prefetch_full_params is True
+            and self._fsdp_graph_order is not None
+            and self._my_fsdp_idx_in_graph is not None
+            and self._my_fsdp_idx_in_graph < len(self._fsdp_graph_order) - 1
```

Comment:
I think the backward prefetch code uses this too, we can probably dedupe into a helper function?

Reply:
I made some refactors and merge the three functions into one `_need_prefetch_full_params`: https://github.com/pytorch/pytorch/pull/78841/files#diff-ffc1ad1876fbf03d814108f6f2f5950a99281b47e0fd77597745f10dc99add86R1684, 
please let me know if you have any suggestions!